modules/git/repo_commit.go

```diff
@@ -198,9 +198,9 @@ func (repo *Repository) GetCommitByPath(relpath string) (*Commit, error) {
 // CommitsRangeSize the default commits range size
 var CommitsRangeSize = 50
 
-func (repo *Repository) commitsByRange(id SHA1, page int) (*list.List, error) {
-	stdout, err := NewCommand("log", id.String(), "--skip="+strconv.Itoa((page-1)*CommitsRangeSize),
-		"--max-count="+strconv.Itoa(CommitsRangeSize), prettyLogFormat).RunInDirBytes(repo.Path)
+func (repo *Repository) commitsByRange(id SHA1, page, pageSize int) (*list.List, error) {
+	stdout, err := NewCommand("log", id.String(), "--skip="+strconv.Itoa((page-1)*pageSize),
```

Comment:
This may preclude a cool technique (I've literally just considered): How do you know if there are more pages if you can't just get a simple count from the start?

If you request one more than you actually want and get it you actually know there's at least one more page. The use of page and pagesize means that only things which can be easily totally enumerated can be paged - an ability to know if there is at least one more thing might be helpful where a pre count can't be easily determined.